vault/seal_manager.go

```diff
@@ -130,14 +130,27 @@ func (sm *SealManager) StorageAccessForPath(path string) StorageAccess {
 }
 
 // SealNamespace seals the barriers of the given namespace and all of its children.
-func (sm *SealManager) SealNamespace(ns *namespace.Namespace) error {
+func (sm *SealManager) SealNamespace(ctx context.Context, ns *namespace.Namespace) error {
 	var errs error
 	sm.barrierByNamespace.WalkPrefix(ns.Path, func(p string, v any) bool {
 		s := v.(SecurityBarrier)
 		if s.Sealed() {
 			return false
 		}
-		err := s.Seal()
+		childNS, err := sm.core.namespaceStore.getNamespaceByPathLocked(ctx, namespace.Canonicalize(p))
+		if err != nil {
+			errs = errors.Join(errs, err)
+		}
+		if childNS == nil {
+			errs = errors.Join(errs, fmt.Errorf("Child Namespace not found for path: %s", p))
+		}
```

Comment:
as it doesn't have to be a child namespace only, I'd call it "descendant"
```suggestion
	descendantNamespace, err := sm.core.namespaceStore.getNamespaceByPathLocked(ctx, namespace.Canonicalize(p))
		if err != nil {
			errs = errors.Join(errs, err)
		}
		if descendantNamespace == nil {
			errs = errors.Join(errs, fmt.Errorf("namespace not found for path: %q", p))
		}
```